server/server.go

```diff
@@ -195,6 +201,12 @@ func newStorage(ctx context.Context, opts ListenOpts) dynamiclistener.TLSStorage
 
 func wrapHandler(handler http.Handler, next http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				logrus.Errorf("Server panicked: %v", err)
+				http.Error(rw, fmt.Sprintf("Server panicked: %v", err), http.StatusInternalServerError)
```

Comment:
This seems like a big change and I'm not sure we want this.

[http.Server](https://pkg.go.dev/net/http#Server)'s already catches errors and logs them with its `ErrorLog` field (that's the field being set by the faulty `errorLog` that we're trying to fix.

I'm not sure we can send a `500` this way because the panic might occur after we start sending the headers (eg: 200).

I'd say let's just fix the error logging instead.